src/GNDStk/type2node.hpp

```diff
@@ -9,37 +9,46 @@ The description of those, seen in reverse, applies here.
 
 In short, these functions are for taking various types, perhaps user-defined,
 and converting them into a node that's destined to be added as a child node.
-*/
 
-#include "GNDStk/type2node/src/detail.hpp"
+At the moment, this file is empty except for comments. We're intentionally
+leaving it here, as a placeholder, in case we decide that we *should* have
+some default type-to-node convert() functions.
+*/
 
 
 
 // -----------------------------------------------------------------------------
-// convert(some containers, Node)
+// convert(some std:: sequence containers, Node)
 // -----------------------------------------------------------------------------
 
-#define GNDSTK_CONVERT(CONTAINER) \
-   template< \
-      class T, class Alloc, \
-      template<class...> class METADATA_CONTAINER, \
-      template<class...> class CHILDREN_CONTAINER \
-   > \
-   void convert( \
-      const std::CONTAINER<T,Alloc> &container, \
-      Node<METADATA_CONTAINER,CHILDREN_CONTAINER> &node \
-   ) { \
-      try { \
-         node.clear().name = "values"; \
-         convert(container, detail::put_pcdata_string(node)); \
-      } catch (const std::exception &) { \
-         log::context("convert(" #CONTAINER ",Node)"); \
-         throw; \
-      } \
-   }
-
-   GNDSTK_CONVERT(deque)
-   GNDSTK_CONVERT(list)
-   GNDSTK_CONVERT(vector)
-
-#undef GNDSTK_CONVERT
+/*
```

Comment:
Comments about what used to be in the code don't need to be there. Now, I have to maintain these comments, which will immediately be out of date when the Pull Request is finished. 

I appreciate the documentation about why this was removed, but it should go in the description of the Pull Request and not in the code itself.

Reply:
I thought about this at the time, but decided to try to formulate the comment in such a way that it would provide a bit more value than just, "we once had these, but no longer."

Conversions between objects in their tree form, and custom objects, are an integral part of GNDStk. I anticipate that we *may* thus have code again in this file someday (now, it's just comments). That, first, is why I kept the file -- as essentially a placeholder.

And, then, given that I kept the file, I wanted to say something about why I *don't* have something that might seem natural to have -- namely, those particular conversions that I removed.

This way, we'll avoid the possibility that somebody might see the file, think "hey, the original author missed this thing that's so natural to have," and then waste time rewriting essentially the same thing that I intentionally removed, before they finally discover the hard way *why* I removed it.

Someone working on the code in the future is more likely to see what's written directly in the code, right in front of their eyes, than something buried in old PR comments from months or years earlier. That's my thought at least.

I supposed I could reword the comment so that it doesn't even say such-and-such content was *once* there, but just says why it isn't there...period.

Reply:
Just taking my previous remark a bit further.... :-) *Good* codes have comments about what they do, and perhaps why. *Great* codes also have, at least in select places, comments about why they *don't* do certain things.

On more than one occasion, I've revisited my own code that I hadn't worked on for years, and thought that something obvious was missing, or that I could do this or that in a simpler way than I was doing it.

Then, after writing code again and running into problems, something in my brain would click, and I'd remember that I dealt with the same issue years before, but had forgotten about it.

So, these days, when I rediscover something in this way (or when I anticipate that I might, as I did here), I write a comment. A comment about what I *don't* have. That'll save someone, possibly myself, some amount of grief down the road. :-)

Very simple example: the comment that begins with "Otherwise..." a few lines up from the bottom of `Node/src/call-child.hpp`. (Although that's not so much about something I don't have, but about why I don't do something that's simpler but in fact wrong.)